community/cypher/cypher/src/test/scala/org/neo4j/cypher/EagerizationAcceptanceTest.scala

```diff
@@ -1439,10 +1470,21 @@ class EagerizationAcceptanceTest extends ExecutionEngineFunSuite with TableDrive
   }
 
 
+  test("matching label on left-hand side and removing same label on right-hand side should not be eager") {
+    createLabeledNode("Lol")
+    createNode()
+    val query = "MATCH (m:Lol), (n) REMOVE n:Lol RETURN count(*)"
```

Comment:
There's no guarantee here that the label scan goes to the left with the cost planner.